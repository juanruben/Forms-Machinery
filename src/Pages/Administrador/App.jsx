import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './Components/Home';
import RegistroHistorico from './Components/RegistroHistorico';
import ClientLists from './Components/Client/ClientLists';
import ClientAdd from './Components/Client/ClientAdd';
import UserLists from './Components/User/UserLists';
import UserAdd from './Components/User/UserAdd';
import FormAdd from './Components/Form/FormAdd';
import FormLists from './Components/Form/FormLists';
import MachineryLists from './Components/Machinery/MachineryLists';
import MachineryAdd from './Components/Machinery/MachineryAdd';

const App = () => (
  <Router>
    <Switch>
      <Route exact path="/administrador/home/" component={Home} />
      <Route exact path="/administrador/registro-historico/" component={RegistroHistorico} />
      <Route exact path="/administrador/cliente/add/" component={ClientAdd} />
      <Route exact path="/administrador/cliente/lists/" component={ClientLists} />
      <Route exact path="/administrador/usuario/add/" component={UserAdd} />
      <Route exact path="/administrador/usuario/lists/" component={UserLists} />
      <Route exact path="/administrador/maquina/add/" component={MachineryAdd} />
      <Route exact path="/administrador/maquina/lists/" component={MachineryLists} />
      <Route exact path="/administrador/formulario/add/" component={FormAdd} />
      <Route exact path="/administrador/formulario/lists/" component={FormLists} />
    </Switch>
  </Router>
);

export default App;
