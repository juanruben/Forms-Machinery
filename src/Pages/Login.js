import React, { Component } from 'react';
import { Container, Row, Col, Form, FormGroup, Button, Alert} from 'reactstrap';


//import Logo from './../Assets/Images/Logo.png';

class Login extends Component {
    constructor(props){
        super(props);
        this.state = {
            usuario: '',
            password: '',
            txtMsj:'',
            AlertClass:'',
        }
    }
	render() {
		return (
            <>
                <Container>
                    <Row>
                        <Col>
                            <div className="content-logo">
                                {/*<img src={Logo} alt="" />*/}
                            </div>
                        </Col>
                    </Row>
                    <Form>
                    
                        <FormGroup row>
                            <Col sm={12}>
                                <input type="text" name="usuario" id="usuario" onChange={this.handleInputChange} value={this.state.usuario || ''} placeholder="Usuario" className="form-control" />
                            </Col>
                        </FormGroup>
                        <FormGroup row>
                            <Col sm={12}>
                                <input type="password" name="password" id="password" onChange={this.handleInputChange} value={this.state.password || ''} placeholder="Password" className="form-control" />
                            </Col>
                        </FormGroup>
                        <FormGroup row>
                            <Col sm={12} className="text-center"> 
                                {this.state.txtMsj !== "" ? <Alert className={'alert '+this.state.AlertClass}>{this.state.txtMsj}</Alert> : "" }
                            </Col>
                        </FormGroup>
                        <FormGroup row>
                            <Col sm={12} className="text-center">
                                <Button onClick={this.setForm} disabled={!this.state.disabledButton} className="btn-orange">Ingresar</Button>
                            </Col>
                        </FormGroup>
                    </Form>
                </Container>
            </>
        )
	}
}

export default Login;