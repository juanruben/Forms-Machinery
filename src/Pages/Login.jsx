import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {
  Container,
  Row,
  Col,
  Form,
  FormGroup,
  Button,
  Alert,
} from 'reactstrap';
import Api from '../Service/Api';
import Logo from '../Assets/images/Logo.png';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      usuario: '',
      password: '',
      msj: '',
      alertclass: '',
      disabledbutton: '',
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.procesar = this.procesar.bind(this);
  }

  async procesar() {
    const { usuario, password } = this.state;
    const dataProcesar = {
      usuario,
      password,
    };
    const response = await Api.Login(dataProcesar);

    if (response.result === 'success') {
      if (response.perfil === 1) {
        window.location.assign('/administrador/');
      } else {
        window.location.assign('/operador/');
      }
    } else {
      this.setState({
        msj: 'Usuario y/o contraseña incorrectos',
        alertclass: 'alert-danger',
      });
    }
  }

  handleInputChange(event) {
    this.setState({ msj: '' });
    const { target } = event;
    const { value } = target;
    const { name } = target;
    this.setState({ [name]: value });
    const { usuario, password } = this.state;
    if (usuario !== '' && password !== '') {
      this.setState({ disabledbutton: true });
    } else {
      this.setState({ disabledbutton: false });
    }
  }

  render() {
    const {
      usuario,
      password,
      msj,
      alertclass,
      disabledbutton,
    } = this.state;

    return (
      <Container>
        <Row>
          <Col md={{ size: 6, offset: 3 }}>
            <div className="container-login box">
              <div className="container__logo">
                <img src={Logo} alt="" />
              </div>
              <Form>
                <FormGroup row>
                  <Col sm={12}>
                    <input type="text" name="usuario" id="usuario" onChange={this.handleInputChange} value={usuario || ''} placeholder="Usuario" className="form-control" />
                  </Col>
                </FormGroup>
                <FormGroup row>
                  <Col sm={12}>
                    <input type="password" name="password" id="password" onChange={this.handleInputChange} value={password || ''} placeholder="Password" className="form-control" />
                  </Col>
                </FormGroup>
                <FormGroup row>
                  <Col sm={12} className="text-center">
                    {msj !== '' ? <Alert className={`alert ${alertclass}`}>{msj}</Alert> : '' }
                  </Col>
                </FormGroup>
                <FormGroup row>
                  <Col sm={12} className="text-center">
                    <Button onClick={this.procesar} className="btn-orange" disabled={disabledbutton}>Entrar</Button>
                  </Col>
                </FormGroup>
                <FormGroup row>
                  <Col sm={12} className="text-center">
                    <Link to="/recuperar" className="link">Olvidó su contraseña</Link>
                  </Col>
                </FormGroup>
              </Form>
            </div>
          </Col>
        </Row>
      </Container>
    );
  }
}
export default Login;
