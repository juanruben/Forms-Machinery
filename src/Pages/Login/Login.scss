.login-container {
  input {
    width: 260px;
  }
}

.link-login {
  color: rgb(122, 122, 122);
  display: block;
  text-align: center;
  margin-bottom: 30px;
  margin-top: 20px;
  text-decoration: underline;

  &:hover {
    color: black;
  }
}
