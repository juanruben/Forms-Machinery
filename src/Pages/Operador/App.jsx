import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './Components/Home';

const App = () => (
  <Router>
    <Switch>
      <Route exact path="/operador/home/" component={Home} />
    </Switch>
  </Router>
);
export default App;
