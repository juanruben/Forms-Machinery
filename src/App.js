import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Login from './Pages/Login';
import Home from './Pages/Home';

class App extends Component {
	render() {
		return (
        <Router>
            <Switch> 
              <Route exact path="/" component={Login}  />
              <Route exact path="/Home" component={Home}  />
            </Switch>
        </Router>
    )
	}
}

export default App;