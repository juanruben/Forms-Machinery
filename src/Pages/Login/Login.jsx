import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { StateContext } from '../../State';
import { validateUsername, validatePassword } from '../../Service/Utils';
import LayoutFullWidth from '../../Layout/LayoutFullWidth/LayoutFullWidth';
import Box from '../../Layout/Box/Box';
import Logo from '../../Components/Logo/Logo';
import Input from '../../Components/Input/Input';
import Button from '../../Components/Button/Button';
import { AlertDialog } from '../../Components/Dialog/Dialog';
import { login } from '../../Service/Api';
import './Login.scss';

class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            data: {},
            errors: {},
            showAlertError: false,
            alertMessage: '',
        };
        this.onChange = this.onChange.bind(this);
        this.onKeyPress = this.onKeyPress.bind(this);
        this.validForm = this.validForm.bind(this);
        this.checkLoggedIn = this.checkLoggedIn.bind(this);
        this.handleSignIn = this.handleSignIn.bind(this);
        this.requestSignIn = this.requestSignIn.bind(this);
        this.signIn = this.signIn.bind(this);
    }

    componentDidMount() {
        this.checkLoggedIn();
        document.title = 'Forms Machinery';
    }

    onChange(event) {
        const { name, value } = event.target;
        const { data, errors } = this.state;
        data[name] = value;
        errors[name] = '';
        this.setState({
            data,
            errors,
        });
    }

    onKeyPress(event) {
        if (event.key === 'Enter') {
            this.handleSignIn();
        }
    }

    checkLoggedIn() {
        const [, dispatch] = this.context;
        dispatch({
            type: 'LOAD_SESSION',
        });
    }

    validForm() {
        const { data } = this.state;
        const { user, password } = data;
        const errors = {};
        let formIsValid = true;

        if (!user || user.trim().length === 0) {
            formIsValid = false;
            errors.user = ['Requerido'];
        } else if (!validateUsername(user)) {
            formIsValid = false;
            errors.user = ['Error de formato'];
        }

        if (!password || password.trim().length === 0) {
            formIsValid = false;
            errors.password = ['Requerido'];
        } else if (!validatePassword(password)) {
            formIsValid = false;
            errors.password = ['Debe tener al menos: 1 mayúscula, 1 minúscula, 1 número'];
        }

        this.setState({
            errors,
        });

        return formIsValid;
    }

    toggleLoading(value) {
        const [, dispatch] = this.context;
        dispatch({
            type: 'SET_LOADING',
            value,
        });
    }

    async requestSignIn() {
        const { data } = this.state;

        await login(data).then((response) => {
            if (response && response.status === 200) {
                const { token, role } = response.data;
                this.signIn(token, role);
            } else {
                this.setState({
                    showAlertError: true,
                    alertMessage: 'Error de conexión',
                });
            }
        }).catch(() => {
            this.setState({
                showAlertError: true,
                alertMessage: 'Datos incorrectos',
            });
        });
        this.toggleLoading(false);
    }

    signIn(token, role) {
        const [, dispatch] = this.context;
        dispatch({
            type: 'SIGN_IN',
            value: { token, role },
        });
    }

    handleSignIn() {
        if (this.validForm()) {
            this.toggleLoading(true);
            this.requestSignIn();
        }
    }

    render() {
        const [{ loggedin, role }] = this.context;
        const {
            showAlertError, alertMessage, errors, data,
        } = this.state;
        const { user, password } = data;


        const path = role === 1 ? '/admin/dashboard' : '/entrada';
        if (loggedin) {
            return (
                <Redirect to={path} />
            );
        }

        return (
            <LayoutFullWidth>
                <Box>
                    <Logo padding={30} maxWidth={150} />
                    <div className="login-container">
                        <Input type="text" name="user" onChange={this.onChange} value={user} icon="fas fa-user-tie" placeholder="Usuario" errors={errors} />
                        <Input type="password" name="password" onChange={this.onChange} onKeyPress={this.onKeyPress} value={password} icon="fas fa-unlock-alt" placeholder="Contraseña" errors={errors} />
                        <div className="login-footer">
                            <Button type="submit" onClick={this.handleSignIn} text="Entrar" />
                        </div>
                    </div>
                    <Link to="/recuperar" className="link-login">Olvidó su contraseña</Link>
                </Box>

                <AlertDialog
                    message={alertMessage}
                    show={showAlertError}
                    onConfirm={() => { this.setState({ showAlertError: false }); }}
                />
            </LayoutFullWidth>
        );
    }
}

Login.contextType = StateContext;

export default Login;
