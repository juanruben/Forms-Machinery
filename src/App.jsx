import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Login from './Pages/Login';
import Recuperar from './Pages/Recover';
import Administrador from './Pages/Administrador/App';
import Operador from './Pages/Operador/App';

const App = () => (
    <Router>
        <Switch>
            <Route exact path="/" component={Login} />
            <Route exact path="/Recuperar" component={Recuperar} />
            <Route path="/administrador/" component={Administrador} />
            <Route path="/operador/" component={Operador} />
        </Switch>
    </Router>
);

export default App;
