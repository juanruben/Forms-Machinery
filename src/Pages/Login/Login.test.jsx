import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { StateProvider } from '../../State';
import { initialState, reducer } from '../../App';
import AppWithLoader from '../../AppWithLoader';
import Login from './Login';

jest.mock('./../../Service/Api');

describe('Login', () => {
    let wrapper;
    let wrapperLogin;
    beforeEach(() => {
        wrapper = mount(
            <StateProvider initialState={initialState} reducer={reducer}>
                <AppWithLoader />
            </StateProvider>,
        );
        wrapperLogin = wrapper.find(Login).at(0);
    });

    it('should render correctly', () => expect(wrapper.debug()).toMatchSnapshot());

    it('should show and hide loading', () => {
        act(() => {
            wrapperLogin.instance().toggleLoading(true);
        });
        wrapper.update();
        expect(wrapper.find('Loader').prop('visible')).toBeTruthy();
        act(() => {
            wrapperLogin.instance().toggleLoading(false);
        });
        wrapper.update();
        expect(wrapper.find('Loader').prop('visible')).toBeFalsy();
    });

    it('should call functions on login with no data', () => {
        const spyValidation = jest.spyOn(wrapperLogin.instance(), 'handleValidation');
        const spyLoader = jest.spyOn(wrapperLogin.instance(), 'toggleLoading');
        act(() => {
            wrapperLogin.instance().handleSignIn();
        });
        expect(spyValidation).toBeCalledTimes(1);
        expect(spyLoader).toBeCalledTimes(0);
    });

    it('should update state', () => {
        wrapperLogin.instance().handleInputChange({ target: { name: 'username', value: 'test' } });
        expect(wrapperLogin.state('username')).toEqual('test');
        wrapperLogin.instance().handleInputChange({ target: { name: 'password', value: 'test2' } });
        expect(wrapperLogin.state('password')).toEqual('test2');
    });

    it('should validate', () => {
        wrapperLogin.setState({ username: 'test', password: 'Test123' });
        expect(wrapperLogin.instance().handleValidation()).toBe(true);
        wrapperLogin.setState({ username: '' });
        expect(wrapperLogin.instance().handleValidation()).toBe(false);
        wrapperLogin.setState({ password: '' });
        expect(wrapperLogin.instance().handleValidation()).toBe(false);
    });

    it('should have a box container', () => {
        expect(wrapper.find('.box-container')).toHaveLength(1);
    });

    it('should have a logo, two inputs and a button', () => {
        expect(wrapper.find('Logo')).toHaveLength(1);
        expect(wrapper.find('Input')).toHaveLength(3);
        expect(wrapper.find('Button')).toHaveLength(1);
    });

    afterEach(() => wrapper.unmount());
});

describe('Login process', () => {
    let wrapper;
    let wrapperLogin;
    beforeEach(() => {
        wrapper = mount(
            <StateProvider initialState={initialState} reducer={reducer}>
                <AppWithLoader />
            </StateProvider>,
        );
        wrapperLogin = wrapper.find(Login).at(0);
    });

    it('should log in', () => {
        wrapperLogin.setState({ username: 'test', password: 'Test123' });
        const spyValidation = jest.spyOn(wrapperLogin.instance(), 'handleValidation');
        const spyLoader = jest.spyOn(wrapperLogin.instance(), 'toggleLoading');

        act(() => {
            wrapperLogin.instance().handleSignIn();
        });
        expect(spyValidation).toBeCalledTimes(1);
        expect(spyLoader).toBeCalledTimes(2);
        wrapper.update();
        expect(wrapper.render().find('Main')).toBeTruthy();
    });

    afterEach(() => wrapper.unmount());
});
