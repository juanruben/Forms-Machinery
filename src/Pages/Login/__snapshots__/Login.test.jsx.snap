// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
"<StateProvider initialState={{...}} reducer={[Function: reducer]}>
  <AppWithLoader>
    <Loader visible={false}>
      <AppRouter>
        <BrowserRouter>
          <Router history={{...}}>
            <Switch>
              <Route exact={true} path=\\"/login\\" component={[function]} location={{...}} computedMatch={{...}}>
                <Login history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                  <LayoutFullWidth>
                    <div className=\\"full-width-container\\">
                      <Box>
                        <div className=\\"box-container\\">
                          <Logo>
                            <div className=\\"logo\\">
                              <img src=\\"Logo.png\\" alt=\\"logo icafal\\" />
                            </div>
                          </Logo>
                          <Input type=\\"text\\" name=\\"username\\" onChange={[Function: bound handleInputChange]} value=\\"\\" icon=\\"fas fa-user-tie\\" placeholder=\\"Usuario\\" errors=\\"\\">
                            <div className=\\"input-bordered-container\\">
                              <i className=\\"fas fa-user-tie\\" />
                              <input className=\\"input-bordered false\\" type=\\"text\\" name=\\"username\\" value=\\"\\" onChange={[Function: bound handleInputChange]} placeholder=\\"Usuario\\" />
                            </div>
                          </Input>
                          <Input type=\\"password\\" name=\\"password\\" onChange={[Function: bound handleInputChange]} value=\\"\\" icon=\\"fas fa-unlock-alt\\" placeholder=\\"Contrase침a\\" errors=\\"\\">
                            <div className=\\"input-bordered-container\\">
                              <i className=\\"fas fa-unlock-alt\\" />
                              <input className=\\"input-bordered false\\" type=\\"password\\" name=\\"password\\" value=\\"\\" onChange={[Function: bound handleInputChange]} placeholder=\\"Contrase침a\\" />
                            </div>
                          </Input>
                          <Input type=\\"text\\" name=\\"rol\\" onChange={[Function: bound handleInputChange]} value={1} icon=\\"fas fa-unlock-alt\\" placeholder=\\"Rol\\" errors=\\"\\">
                            <div className=\\"input-bordered-container\\">
                              <i className=\\"fas fa-unlock-alt\\" />
                              <input className=\\"input-bordered false\\" type=\\"text\\" name=\\"rol\\" value={1} onChange={[Function: bound handleInputChange]} placeholder=\\"Rol\\" />
                            </div>
                          </Input>
                          <Button type=\\"submit\\" onClick={[Function: bound handleSignIn]} text=\\"Entrar\\" disabled={false}>
                            <button type=\\"submit\\" className=\\"main-button\\" onClick={[Function: bound handleSignIn]} disabled={false}>
                              Entrar
                            </button>
                          </Button>
                          <Link to=\\"/recuperar\\" className=\\"link-login\\">
                            <LinkAnchor className=\\"link-login\\" href=\\"/recuperar\\" navigate={[Function: navigate]}>
                              <a className=\\"link-login\\" href=\\"/recuperar\\" onClick={[Function: onClick]}>
                                Olvid칩 su contrase침a
                              </a>
                            </LinkAnchor>
                          </Link>
                        </div>
                      </Box>
                    </div>
                  </LayoutFullWidth>
                </Login>
              </Route>
            </Switch>
          </Router>
        </BrowserRouter>
      </AppRouter>
    </Loader>
  </AppWithLoader>
</StateProvider>"
`;
